docs/guidelines/creating-new-model.md

```diff
@@ -0,0 +1,196 @@
+
+The REST representations are just using the `@commercetools/platform-sdk`.
+This is a package which is auto-generated with the latest versions of the commercetools REST APIs and already contains all the Typescript interfaces describing the models.
+
+On the other hand, the GraphQL representation can use the previous ones as a blueprint, but most likely will use different types for some of their properties (specially the nested models).
+
+## Configuring fields
+
+The next step can involve creating the `fields-config` file where we need to implement the population of the default values we want the test data model to have.
+
+Since we have two representations for it (REST and GraphQL), we need to export two objects.
+
+Here's an example:
+
+```ts
+import {
+  Address,
+  ClientLogging,
+  LocalizedString,
+} from '@commercetools-test-data/commons';
+import {
+  fake,
+  sequence,
+  type TModelFieldsConfig,
+} from '@commercetools-test-data/core';
+import { createRelatedDates } from '@commercetools-test-data/utils';
+import { roles } from './constants';
+import type { TChannelGraphql, TChannelRest } from './types';
+
+const [getOlderDate, getNewerDate] = createRelatedDates();
+
+const commonFieldsConfig = {
+  id: fake((f) => f.string.uuid()),
+  version: sequence(),
+  createdAt: fake(getOlderDate),
+  createdBy: fake(() => ClientLogging.random()),
+  lastModifiedAt: fake(getNewerDate),
+  lastModifiedBy: fake(() => ClientLogging.random()),
+  key: fake((f) => f.lorem.slug(2)),
+  roles: [roles.Primary],
+  name: null,
+  description: null,
+  address: fake(() => Address.random()),
+  reviewRatingStatistics: null,
+  custom: null,
+  geoLocation: null,
+};
+export const restFieldsConfig: TModelFieldsConfig<TChannelRest> = {
+  fields: {
+    ...commonFieldsConfig,
+  },
+};
+export const graphqlFieldsConfig: TModelFieldsConfig<TChannelGraphql> = {
+  fields: {
+    ...commonFieldsConfig,
+    __typename: 'Channel',
+    nameAllLocales: null,
+    descriptionAllLocales: null,
+  },
+  postBuild: (model) => {
+    const name = model.nameAllLocales
+      ? LocalizedString.resolveGraphqlDefaultLocaleValue(model.nameAllLocales)
+      : undefined;
+    const description = model.descriptionAllLocales
+      ? LocalizedString.resolveGraphqlDefaultLocaleValue(
+          model.descriptionAllLocales
+        )
+      : undefined;
+    return {
+      name,
+      description,
+    };
+  },
+};
+```
+
+Probably you have spotted the `postBuild` property in the GraphQL config object.
+
+This is an optional and we should only use if there are any dependencies among the data model fields.
+In this case, `name` and `description` depend on the values from `nameAllLocales` and `descriptionAllLocales` but we don't know them in advance but only once the data model has been built. The `postBuild` callback allows to manipulate the built object before it's returned to the consumer.
```

Comment:
Nit: Small grammatical cleanup.
```suggestion
This is optional and we should only use it if there are any dependencies among the data model fields.
In this case, `name` and `description` depend on the values from `nameAllLocales` and `descriptionAllLocales`. We won't know the values in advance, but will once the data model has been built. The `postBuild` callback allows us to manipulate the built object before it's returned to the consumer.
```

Reply:
Thanks for the suggestion, Sarah 🙇 

Applied here: 783806a087a7f9ab2bc517073b570d0d4dab9866